Modernize for-loops in caffe2/torch (#50797)

r-barnes opened the pull request:
Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/50797

Modernize for-loops throughout caffe2/ subdirs to use ranged-loops where possible (all `.cpp` files were examined).

```
find caffe2/ -iname "*.cpp" > /home/rbarnes/files
buck run mode/opt foundation/clangr:clangr_local -- -j 10 --file=/home/rbarnes/files --multi --apply-replacements=true tidy '--checks=-*,modernize-loop-convert'
```

Test Plan: Sandcastle tests

Differential Revision: D26585065

pritamdamania87 commented:
The distributed changes look good to me, although it looks like there are some CI failures. Maybe someone from the JIT team can sign off on the JIT changes

cc @suo

r-barnes committed:
Modernize for-loops in caffe2/torch (#52618)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/52618

Pull Request resolved: https://github.com/pytorch/pytorch/pull/50797

Modernize for-loops throughout caffe2/ subdirs to use ranged-loops where possible (all `.cpp` files were examined).

```
find caffe2/ -iname "*.cpp" > /home/rbarnes/files
buck run mode/opt foundation/clangr:clangr_local -- -j 10 --file=/home/rbarnes/files --multi --apply-replacements=true tidy '--checks=-*,modernize-loop-convert'
```

Test Plan: Sandcastle tests

Reviewed By: suo

Differential Revision: D26585065

fbshipit-source-id: c1c478de2c144dc2e93da24fc4ef22dbaa72d034
